--- a/arch/arm64/boot/dts/qcom/msm8916.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916.dtsi
@@ -1520,7 +1520,7 @@
 		tcsr: syscon@1937000 {
-			compatible = "qcom,tcsr-msm8916", "syscon";
+			compatible = "qcom,tcsr-msm8916", "syscon", "simple-mfd";
 			reg = <0x01937000 0x30000>;
 		};
 
@@ -1524,6 +1524,18 @@
 			reg = <0x01937000 0x30000>;
 		};
 
+		reboot-mode@193d100 {
+			compatible = "syscon-reboot-mode";
+			syscon = <&tcsr>;
+			offset = <0xd100>;
+			mode-normal = <0x77665501>;
+			mode-bootloader = <0x77665500>;
+			mode-recovery = <0x77665502>;
+			mode-fastboot = <0x77665500>;
+			mode-edl = <0x77665503>;
+			mode-download = <0x646f776e>;
+		};
+
 		tcsr_mutex: hwlock@1905000 {
 			compatible = "qcom,tcsr-mutex";
 			reg = <0x01905000 0x20000>;
